packages/core/upload/admin/src/pages/App/components/BulkMoveButton.js

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import { useIntl } from 'react-intl';
+import { Button } from '@strapi/design-system/Button';
+import Folder from '@strapi/icons/Folder';
+import { getAPIInnerErrors } from '@strapi/helper-plugin';
+
+import { BulkMoveDialog } from '../../../components/BulkMoveDialog';
+import { AssetDefinition, FolderDefinition } from '../../../constants';
+import { useBulkMove } from '../../../hooks/useBulkMove';
+import { getTrad } from '../../../utils';
+
+export const BulkMoveButton = ({ selected, onSuccess }) => {
+  const { formatMessage } = useIntl();
+  const [showConfirmDialog, setShowConfirmDialog] = useState(false);
+  const [dialogErrors, setDialogErrors] = useState(null);
+  const { move } = useBulkMove();
+
+  const handleConfirmMove = async ({ moved, destinationFolderId } = {}) => {
+    try {
+      if (moved) {
+        await move(destinationFolderId, selected);
+        onSuccess();
+      }
+
+      setShowConfirmDialog(false);
+      // eslint-ignore-next-line no-empty
+    } catch (error) {
+      const errors = getAPIInnerErrors(error, { getTrad });
+      const formikErrors = Object.entries(errors).reduce((acc, [key, error]) => {
+        acc[key || 'destination'] = error.defaultMessage;
+
+        return acc;
+      }, {});
+
+      if (!isEmpty(formikErrors)) {
+        setDialogErrors(formikErrors);
+      }
+    }
+  };
+
+  useEffect(() => {
```

Comment:
I tried to remove this one and everything seems to work without it (success or error), am I missing something?

Reply:
I'll check and think it is necessary because the modal could have been closed at this stage. Good catch.

Reply:
Done in b47e510918c46f6b93905e3893491beb737ef6b1